source/adios2/engine/bp5/BP5Writer.cpp

```diff
@@ -1217,6 +1323,35 @@ void BP5Writer::InitAggregator()
         DataWritingComm = &m_AggregatorEveroneWrites.m_Comm;
         m_Aggregator = static_cast<aggregator::MPIAggregator *>(&m_AggregatorEveroneWrites);
     }
+    else if (m_Parameters.AggregationType == (int)AggregationType::DataSizeBased)
+    {
+        // Close() is a no-op if never initialized
+        std::cout << "InitAggregator() - DataSizeBased: Closing and re-opening MPIChain"
+                  << std::endl;
+        m_AggregatorDataSizeBased.Close();
+        m_AggregatorDataSizeBased.InitSizeBased(DataSize, m_Parameters.NumSubFiles, m_Comm);
+        m_IAmDraining = m_AggregatorEveroneWrites.m_IsAggregator;
+        m_IAmWritingData = true;
+        DataWritingComm = &m_AggregatorDataSizeBased.m_Comm;
+        m_Aggregator = static_cast<aggregator::MPIAggregator *>(&m_AggregatorDataSizeBased);
+
+        /* comm for Aggregators only.
+         *  We are only interested in the chain of rank 0s
+         */
+        int color = m_Aggregator->m_Comm.Rank();
+        if (m_WriterStep > 0)
+        {
+            m_CommAggregators.Free("freeing aggregators comm for data-size based aggregation");
+        }
+        else
+        {
+            // Stuff we only want to do on the first timestep
+            m_SubstreamDataPos.resize(m_Parameters.NumSubFiles);
+        }
+        m_CommAggregators =
```

Comment:
It appeared to me that `m_CommAggregators` was split off from the world, and then never used anywhere, so I decided to re-purpose it.  Hopefully that's acceptable and I didn't misread the code.